### ISSUE TITLE
1.0 Beta 1 -  "No such module 'ApolloAPI'"

### ISSUE THREAD

**Issue by scottasoutherland** [apollographql/apollo-ios#2418]
## Bug report

I'm trying to use the 1.0 Beta 1 release with Cocoapods. I get a compilation error in `GraphQLResult` which says `No such module 'ApolloAPI'` . When looking at the pod contents, i see ApolloAPI is not included. It looks like it is included in the SPM file but not in the PodSpec. 

## Versions

Please fill in the versions you're currently using: 

- `apollo-ios` SDK version: 1.0 Beta 1
- Xcode version:  13.4
- Swift version:  5.6
- Package manager: Cocoapods

## Steps to reproduce

- Setup a project using cocoapods. 
- Add the following line to your podfile: `pod 'Apollo', :git => 'https://github.com/apollographql/apollo-ios.git', :branch => '1.0.0-beta.1'`
- Run `pod install`
- Build the project.

**Comment by calvincestari** [comment]
Hi @scottasoutherland, thanks for the bug report. This looks like a recent bug on our part, I'll get it resolved ASAP.

**Comment by calvincestari** [comment]
@scottasoutherland - I believe #2420 should resolve your issue. You can test that branch with your project by changing your Podfile from targeting `1.0.0-beta.1` to `fix/issue-2418`. Once that is merged into `release/1.0` we'll consider issuing a `1.0.0-beta.2` release; we've got a few things queued up for the next release.

**Comment by scottasoutherland** [comment]
> Hey @calvincestari , thank you for the speedy response and fix. The fix you posted _does_ fix the compilation of the Apollo pod, however all of my generated code from the apollo code generation still has `import ApolloAPI` at the top, so my project will not compile. Let me know if you'd like me to file a separate bug for that or if it can stay as part of this bug.

Perhaps it is worth noting, that our code generation project use SPM and our client project which uses the generated files, uses cocoapods.

**Comment by calvincestari** [comment]
Ah, OK. Our codegen templates need to change to wrap the ApolloAPI import in the `#if COCOAPODS` block too then. I'll get that fixed up and added to that PR.

Thanks for letting me know, we certainly appreciate all the feedback on 1.0.

**Comment by scottasoutherland** [comment]
As i mentioned our codegen is actually a separate project which uses SPM because we have our graphQL libraries in a separate repo. So i'm not 100% sure if that change would fix it. It sounds like perhaps that use case is not going to be supported going forward if SPM treats ApolloAPI as a separate dependency but CocoaPods treats it as part of the same library?

**Comment by calvincestari** [comment]
I don't think I completely understand your configuration yet but there's a couple things to try separate here:
1. **Code generation 'project'** - in 0.x releases and documentation, this could either be a [Build Phase](https://www.apollographql.com/docs/ios/installation/#5-add-a-code-generation-build-step) of the main project or an actual separate project/target described in the [Swift Scripting approach](https://www.apollographql.com/docs/ios/swift-scripting/). This could be done with SPM and the [iOSCodegenTemplate](https://github.com/apollographql/iOSCodegenTemplate), as I think you're describing. This is going to be replaced by a [CLI tool](https://github.com/apollographql/apollo-ios/tree/release/1.0/CodegenCLI) that you can use with a local JSON configuration file; the `iOSCodegenTemplate`-style way of will remain because you can always create your own project with any highly custom generation steps needed.
2. The generated files - this output is configured by your [`ApolloCodegenConfiguration.output`](https://github.com/apollographql/apollo-ios/blob/release/1.0/Sources/ApolloCodegenLib/ApolloCodegenConfiguration.swift#L160) configuration. The documentation in that file is quite comprehensive but the important thing to note is that the are settings for schema types, operation files, and test mocks. These can all be in a single module or each in different modules. Define the output for your operation files for the type of dependency module in which they will be used.
3. Client project - this is the product that ships and is usually the consumer of the generated models based on your operation files. To reiterate, it's important that the configuration of [`ApolloCodegenConfiguration.output.operations`](https://github.com/apollographql/apollo-ios/blob/release/1.0/Sources/ApolloCodegenLib/ApolloCodegenConfiguration.swift#L184) match the dependency management system you're using here.

We haven't intentionally removed any supported configurations so if this can't be resolved through configuration we'll need to take another look. Can you share a bit more about your file structure, configuration file or even a sample project demonstrating your setup. We'll do our best to get this resolved for you.

**Comment by scottasoutherland** [comment]
For 1. We have a separate project entirely for codegen. It's actually even in a different repo. It uses swift package manager. 
For 3. Our client which consumes Apollo and the generated code uses Cocoapods. Hence if the codegen relied on whether it was built using SPM to decide whether to import ApolloAPI, it wouldn't matter, because the generated code is is used in a different environment than it was built in. 

I'm unclear on how the  `output.operations` would make a difference. I would think that [ModuleType](https://github.com/apollographql/apollo-ios/blob/ba6bcae51561aec56f52a9ac25603751a1d71984/Sources/ApolloCodegenLib/ApolloCodegenConfiguration.swift#L217) could be used for that. But we are already using `other` there, but perhaps there is another issue causing codegen not to respect that for imports / namespacing of ApolloAPI

**Comment by calvincestari** [comment]
Hey @scottasoutherland - apologies for taking so long to get back to you, I was away on vacation last week.

Taking another look at this I don't think the output configuration can resolve it. https://github.com/apollographql/apollo-ios/pull/2420 will resolve only the apollo-ios files but for generated code the [schema template](https://github.com/apollographql/apollo-ios/blob/release/1.0/Sources/ApolloCodegenLib/Templates/TemplateRenderer.swift#L98) is not configured to generate the import header correctly. @AnthonyMDev and I will be talking about this today.
